### ISSUE TITLE
Linprog(method='simplex') produces incorrect starting basis.

### ISSUE THREAD

**Issue by Kai-Striega** [scipy/scipy#9008]
The simplex method produces an incorrect starting basis when converting the problem into cannonical form. The column corresponding to a basic variable should contain zeros except for a single entry which should be one. Allowing the system `A * x_b = b` to be solved exactly. 

The below example could produce a valid set of basic  variables [5, 7, 8, 9, 10] but instead produces basic variables [8, 9, 4, 10, 5] which corresponds the the basis:

```python
[[ 1. -0. -0. -0. -0.]
 [-0.  1. -1. -0. -0.]
 [ 0.  0.  0.  0.  1.]
 [-0. -0. -0.  1. -0.]
 [ 0.  0.  0.  0.  0.]]
```
But there are multiple problems with this formulation,  which should not be possible in the  simplex method

1. Row two contains multiple non-zero values.
2. Column three contains a negative 1.
3. Row five contains all zeros.
 
### Reproducing code example:
```python
# source: test_nontrivial_problem from LinprogCommonTests
from scipy.optimize import linprog, linprog_verbose_callback

c = [-1, 8, 4, -6]
A_ub = [[-7, -7, 6, 9],
        [1, -1, -3, 0],
        [10, -10, -7, 7],
        [6, -1, 3, 4]]
b_ub = [-3, 6, -6, 6]
A_eq = [[-10, 1, 1, -8]]
b_eq = [-4]

res = linprog(
    c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
    callback=linprog_verbose_callback)
```

### Relevant Output
```python
"--------- Initial Tableau - Phase 1 ----------"

[[ 10.0   -1.0   -1.0    8.0  -0.0  -0.0  -0.0  -0.0   1.0  -0.0  -0.0   4.0]
 [  7.0     7.0    -6.0  -9.0  -1.0  -0.0  -0.0  -0.0  -0.0   1.0  -0.0   3.0]
 [  1.0   -1.0    -3.0    0.0   0.0   1.0   0.0   0.0   0.0   0.0   0.0   6.0]
 [ -10.0  10.0   7.0   -7.0  -0.0  -0.0  -1.0  -0.0  -0.0  -0.0   1.0   6.0]
 [  6.0   -1.0     3.0    4.0   0.0   0.0   0.0   1.0   0.0   0.0   0.0   6.0]
 [ -1.0    8.0     4.0   -6.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0]
 [ -7.0 -16.0     0.0    8.0   1.0   0.0   1.0   0.0   0.0   0.0   0.0 -13.0]]

Pivot Element: T[1, 1]

Basic Variables: [ 8  9  4 10  5]

Current Solution:
x =  [  0.0   0.0   0.0   0.0]

Current Objective Value:
f =  13.0
```

### Scipy/Numpy/Python version information:
```
platform linux -- Python 3.6.5, pytest-3.6.2, py-1.5.4, pluggy-0.6.0
```

**Comment by Kai-Striega** [comment]
@mdhaber The error occurs for negative values in the bub constraint. The current simplex method adds the slack and surplus variables **then** multiplies the row by -1 causing the surplus variables to become negative slack variables (and then no longer basic variables.). 

Fixing this now returns a valid (invertible) basis however causes bugs later in the method. I'm hesitant to dig further into these given that all the current tests are passing and focus on the current bugs. The bugs originate (I think) numerical issues which appear inherint to the simplex method (see [1]). 

Some considerations given by [2] could help, the most promising being an implementation of the revised simplex method (an added bonus: it avoids pivoting, requires less memory and fewer iterations). Most commerical solvers implement some form of this method. I've played around with the naive implementation in [3], but there are still a few too many bugs. Do you think it would be better to focus on fixing the current bugs or implementing a revised simplex solver?

[1] Ogryczak, W. (1988). The simplex method is not always well behaved. Linear Algebra and its Applications, 109, 41-57.
[2] Klotz, E., & Newman, A. M. (2013). Practical guidelines for solving difficult linear programs. Surveys in Operations Research and Management Science, 18(1-2), 1-17.
[3] Forrest, J. J., & Tomlin, J. A. (1972). Updated triangular factors of the basis to maintain sparsity in the product form simplex method. Mathematical programming, 2(1), 263-278.

**Comment by mdhaber** [comment]
@Kai-Striega I've wanted to implement a revised simplex solver myself. The problem has been that SuperLU does not support efficient rank-one updates (e.g. Bartels-Golub) and I haven't been able to hack it to do so, despite considerable effort, due to #7700.  I didn't think that the revised simplex solver would be very fast without efficient rank-one updates, so I didn't move with it. Let me know if you agree. If you find a solution (or if I'm wrong in thinking that efficient rank-one updates are needed) it would be great to work on a revised simplex solver with you!

**Comment by Kai-Striega** [comment]
@mdhaber I'll have to read up on the efficiencies/practical implementations of different methods before I can make a proper decision. I'll be on holiday for the next week after which I have more time to work on this. Also shouldn't the question be if the implementation is better than what we currently have?

**Comment by mdhaber** [comment]
@Kai-Striega Well, yes, better than what we currently have is better but it will only happen if it's considered worth the effort. I didn't consider it worth the effort at the time; I was going to wait for SuperLU to be fixed. But I suppose we could implement the revised simplex with provisions for using an efficient update (instead of full LU decomposition) in the future.

**Comment by Kai-Striega** [comment]
Closed via #9145
